#115 1.2 usd viewer features (#123)
Inbibo/shift_docs: docs/integration_resources/resources/USD/usd.md

These two plugins together allow users to efficiently visualize, explore, and interact with USD-based workflows, making Shift a powerful tool for managing complex 3D scenes and assets.

### USD Outliner

> The USD Outliner plugin displays the hierarchy of USD stages sourced from specific nodes (UsdStageOpen & UsdStageNew) in the active board workflow. It provides a tree view of the USD scene, allowing users to explore, select, and interact with USD prims. Selecting a prim in the outliner works in conjunction with the USD Viewer plugin for hierarchical navigation. Double-clicking a prim creates a UsdPrimGet operator linked to the selected prim for further manipulation. For more information on working with USD stages, refer to the section on [USD Catalogs](#catalogs).

Reviewer: What about UsdStage? We do have that node, right?

Author: we do but it is not picked up by the plugin

Author: Well, adding the UsdStage node might cause duplication issues with the same stage being displayed in multiple tabs of the viewer. I don't know whether this might cause problems. Also displaying a stage already loaded in maya in the Usd Viewer as well does not make a lot of sense I think.

What I mean is in general my opinion is that adding the UsdStage node will cause more issues than benefits at this point.

Reviewer: Ok, I see what you mean.
I think that if no one makes a request about this, I think we can keep the current implementation.

      <figcaption>USD Viewer Plugin</figcaption>
</figure>

> The plugin integrates some other features.

Reviewer: I wouldn't add this line. It appears odd on the website, and I don't see it as necessary as long as the features are organized under sub-titles.

```suggestion
```

#### Playback Animated Stages

> When the opened stage contains authored Time Code ranges, the plugin will automatically display them in the timeline widget in the bottom and a *Play* button will appear. This allows users to scrub the stage animation manually or directly playback the entire animation. The plugin will playback at the FPS rate authored in the stage file or default at 24 FPS if no valid authoring is found.

Reviewer: ```suggestion
If the opened stage contains authored Time Code ranges, the plugin will automatically display them in the timeline widget at the bottom, and a *Play* button will appear. This allows users to scrub the stage animation manually or to directly playback the entire animation. The plugin will playback at the FPS rate authored in the stage file or default at 24 FPS if no valid authoring is found.
```

<figure>
      <img src="images/usd_viewer_timeline.gif" alt="UI">
      <figcaption>USD Viewer Plugin Timeline Playback</figcaption>
</figure>

#### Grab Viewport AOV

> Any stage viewport can be linked to a `UsdViewportGrab` node. This linking allows users to send the information from the current Usd Viewer buffer to the Shift workflow as an image.

Reviewer: I don't see other nodes being quoted in this doc. Do you think we should do it? Usually, we link to the node's documentation. I think we can do that here, as it is the first time to been mentioned
```suggestion
Any stage viewport can be linked to a [UsdViewportGrab](#link) node. This linking allows users to send the information from the current Usd Viewer buffer to the Shift workflow as an image.

```

Author: done

<figure>
      <img src="images/usd_viewer_viewport_grab.gif" alt="UI">
      <figcaption>USD Viewer Plugin Viewport Grab</figcaption>
</figure>

> As shown in the figure above, new nodes can be linked using the *Link* menu in the plugin menu bar. When the viewer is linked, the menu name will switch from *Link* to *Linked*. The linked nodes will receive new viewport snapshots whenever the user: pans, zooms or manually changes the current frame.

Reviewer: ```suggestion
As shown in the figure above, new UsdViewportGrab nodes can be linked using the *Link* menu in the plugin menu bar. When the viewer is linked, the menu name will switch from *Link* to *Linked*. Linked nodes will receive new viewport snapshots whenever the user pans, zooms or manually changes the current frame.

```

> The `UsdViewportGrab` node allows to select the desired aov or aspect ratio for the snapshot.

Reviewer: ```suggestion
The `UsdViewportGrab` node allows selecting the desired AOV or aspect ratio for the snapshot.

```

>[!NOTE]

> > Linking the viewer to a node will cause a slight computational overhead to send the information to the workflow. This might cause a decrease in the smoothness of the interaction with the viewer. However, the linking status updates can be temporarly disabled by deselecting the *Active* option in the *Link* menu.

Reviewer: ```suggestion
> Linking the viewer to a node will cause a slight computational overhead to send the information to the workflow. This might cause a decrease in the smoothness of the interaction with the viewer. However, the linking status updates can be temporarily disabled by deselecting the *Active* option in the *Link* menu.

```
